.github/workflows/test.yml

```diff
@@ -30,21 +30,40 @@ jobs:
           enable-cache: true
           python-version: ${{ matrix.python-version }}
           version: ${{ env.UV_VERSION }}
-      - name: Install mandatory dependencies
-        run: uv sync --frozen
       - name: Run tests not requiring optional dependencies
         run: uv run pytest -m 'not require_optional_deps' -n logical --dist loadgroup --junit-xml test-results.xml
-      - name: Install optional dependencies
-        run: uv sync --frozen --group plugin-test
       - name: Run tests requiring optional dependencies
-        run: uv run pytest -m 'require_optional_deps' -n logical --dist loadgroup --junit-xml test-results.xml
+        run: uv run --group plugin-test pytest -m 'require_optional_deps' -n logical --dist loadgroup --junit-xml test-results.xml
       - name: Upload Test Results
         if: always()
         uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4
         with:
           name: Test Results (Python ${{ matrix.python-version }} ${{ matrix.operating-system }})
           path: test-results.xml
 
+  test-scripts:
+    name: Test scripts
+    runs-on: ubuntu-latest
+    env:
+      PYTHON_VERSION: '3.13' # renovate: datasource=python-version depName=python
```

Comment:
We should only need to test these scripts using the production environment's system (Ubuntu) and Python version (3.13).

I don't think it's necessary to run them on all systems and Python versions.